src/Interpreters/TreeRewriter.cpp

```diff
@@ -809,6 +830,8 @@ TreeRewriterResultPtr TreeRewriter::analyzeSelect(
 
     normalize(query, result.aliases, settings);
 
+    removeUnusedAliasesFromWithClause(select_query);
```

Comment:
Do we need to call `removeUnusedAliasesFromWithClause` in `TreeRewriter::analyze` after `normalize` ?

Reply:
Nope, it's for ASTSelect WITH clause exclusively.